moveit_ros/moveit_servo/src/cpp_interface_demo/servo_cpp_interface_demo.cpp

```diff
@@ -0,0 +1,180 @@
+
+// Servo
+#include <moveit_servo/servo_parameters.cpp>
+#include <moveit_servo/servo.h>
+#include <moveit/planning_scene_monitor/planning_scene_monitor.h>
+
+using namespace std::chrono_literals;
+
+int main(int argc, char** argv)
+{
+  rclcpp::init(argc, argv);
+  static const rclcpp::Logger LOGGER = rclcpp::get_logger("moveit_servo.servo_demo_node.cpp");
+
+  // ROS objects
+  rclcpp::NodeOptions node_options;
+  node_options.use_intra_process_comms(true);
+  auto node = std::make_shared<rclcpp::Node>("servo_demo_node", node_options);
+  auto executor = std::make_shared<rclcpp::executors::MultiThreadedExecutor>();
+
+  // Get Servo Parameters
+  auto servo_parameters = std::make_shared<moveit_servo::ServoParameters>();
+  if (!moveit_servo::readParameters(servo_parameters, node, LOGGER))
+  {
+    RCLCPP_ERROR(LOGGER, "Could not get parameters");
+    return -1;
+  }
+
+  // Create the planning_scene_monitor
+  auto tf_buffer = std::make_shared<tf2_ros::Buffer>(node->get_clock());
+  auto planning_scene_monitor = std::make_shared<planning_scene_monitor::PlanningSceneMonitor>(
+      node, "robot_description", tf_buffer, "planning_scene_monitor");
+
+  // Get the planning_scene_monitor to publish scene diff's for RViz visualization
+  if (planning_scene_monitor->getPlanningScene())
+  {
+    planning_scene_monitor->startStateMonitor("/joint_states");
+    planning_scene_monitor->setPlanningScenePublishingFrequency(25);
+    planning_scene_monitor->startPublishingPlanningScene(planning_scene_monitor::PlanningSceneMonitor::UPDATE_SCENE,
+                                                         "/moveit_servo/publish_planning_scene");
+    planning_scene_monitor->startSceneMonitor();
+  }
+  else
+  {
+    RCLCPP_ERROR(LOGGER, "Planning scene not configured");
+    return -1;
+  }
+
+  // Pause for RViz to come up..
+  rclcpp::sleep_for(std::chrono::seconds(4));
+
+  // Create collision object, in the way of servoing
+  moveit_msgs::msg::CollisionObject collision_object;
+  collision_object.header.frame_id = "panda_link0";
+  collision_object.id = "box";
+
+  shape_msgs::msg::SolidPrimitive box;
+  box.type = box.BOX;
+  box.dimensions = { 0.1, 0.4, 0.1 };
+
+  geometry_msgs::msg::Pose box_pose;
+  box_pose.position.x = 0.6;
+  box_pose.position.y = 0.0;
+  box_pose.position.z = 0.6;
+
+  collision_object.primitives.push_back(box);
+  collision_object.primitive_poses.push_back(box_pose);
+  collision_object.operation = collision_object.ADD;
+
+  moveit_msgs::msg::PlanningSceneWorld psw;
+  psw.collision_objects.push_back(collision_object);
+
+  moveit_msgs::msg::PlanningScene ps;
+  ps.world = psw;
+
+  // Publish the collision object to the planning scene
+  auto scene_pub = node->create_publisher<moveit_msgs::msg::PlanningScene>("/planning_scene", 10);
+  scene_pub->publish(ps);
+
+  // Create Servo and start it
+  moveit_servo::Servo servo(node, servo_parameters, planning_scene_monitor);
+  while (!servo.waitForInitialized())
```

Comment:
Put a sleep in the loop and rclcpp::ok() as a terminating condition

Reply:
Done. The `waitForInitialized` includes a sleep (as a wait for message)